planetiler-core/src/main/java/com/onthegomap/planetiler/util/LoopLineMerger.java

```diff
@@ -360,6 +419,24 @@ public void remove() {
       to.removeEdge(reversed);
     }
 
+    double angleTo(Edge other) {
+      if (!from.equals(other.from)) {
+        // Todo raise some error, the edges don't start at same coordinate
+        System.out.println("error start point mismatch");
+        return 0.0;
+      }
+      if (coordinates.size() < 2 || other.coordinates.size() < 2) {
+        // todo no real linestring...
+        System.out.println("error lines are points");
+        return 0.0;
+      }
+
+      double angle = Angle.angle(coordinates.get(0), coordinates.get(1));
+      double angleOther = Angle.angle(other.coordinates.get(0), other.coordinates.get(1));
```

Comment:
FYI you can use coordinates.getFirst() and coordinates.getLast() utilities as shortcuts to get 0 and size-1